src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -21,59 +58,803 @@ const ActorInit En_Dodongo_InitVars = {
+    if (this->actionState > DODONGO_DEATH) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colliderBody.base);
+    }
+    if ((this->actionState >= DODONGO_IDLE) && EnDodongo_AteBomb(this, globalCtx)) {
+        EnDodongo_SetupSwallowBomb(this);
+    }
+    if ((this->actionState == DODONGO_BREATHE_FIRE) && (29.0f < this->skelAnime.animCurrentFrame) &&
+        (this->skelAnime.animCurrentFrame < 43.0f)) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colliderAT.base);
+    }
+    this->actor.posRot2.pos.x = this->actor.posRot.pos.x + Math_Sins(this->actor.shape.rot.y) * -30.0f;
+    this->actor.posRot2.pos.y = this->actor.posRot.pos.y + 20.0f;
+    this->actor.posRot2.pos.z = this->actor.posRot.pos.z + Math_Coss(this->actor.shape.rot.y) * -30.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F9A80.s")
+s32 EnDodongo_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                               Actor* thisx) {
+    EnDodongo* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F9AF8.s")
+    if ((limbIndex == 0xF) || (limbIndex == 0x10)) {
+        Matrix_Scale(this->bodyScale.x, this->bodyScale.y, this->bodyScale.z, MTXMODE_APPLY);
+    }
+    return 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F9C3C.s")
+void EnDodongo_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    static Vec3f legOffsets[3] = {
+        { 1100.0f, -700.0f, 0.0f },
+        { 0.0f, 0.0f, 0.0f },
+        { 2190.0f, 0.0f, 0.0f },
+    };
+    Vec3f tailTipOffset = { 3000.0f, 0.0f, 0.0f };
+    Vec3f baseOffset = { 0.0f, 0.0f, 0.0f };
+    s32 i;
+    Vec3f hardTris0Vtx[3];
+    Vec3f hardTris1Vtx[3];
+    Vec3f hardTris2Vtx[3];
+    Vec3f tailTip;
+    Vec3f tailBase;
+    EnDodongo* this = THIS;
+    Vec3f hardTris0VtxOffset[] = {
+        { -300.0f, -2500.0f, 0.0f },
+        { -300.0f, 1200.0f, -2700.0f },
+        { 3000.0f, 1200.0f, 0.0f },
+    };
+    Vec3f hardTris1VtxOffset[] = {
+        { -300.0f, -2500.0f, 0.0f },
+        { -300.0f, 1200.0f, 2700.0f },
+        { 3000.0f, 1200.0f, 0.0f },
+    };
+    Vec3f hardTris2VtxOffset[] = {
+        { -600.0f, 1200.0f, -2800.0f },
+        { -600.0f, 1200.0f, 2800.0f },
+        { 3000.0f, 1200.0f, 0.0f },
+    };
+    Vec3f mouthOffset = { 1800.0f, 1200.0f, 0.0f };
+    Vec3f headOffset = { 1500.0f, 300.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F9CA0.s")
+    func_800628A4(limbIndex, &this->colliderBody);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Dodongo/func_809F9DC8.s")
+    switch (limbIndex) {
+        case 2:
+            if ((this->actionState == DODONGO_BREATHE_FIRE) && (29.0f < this->skelAnime.animCurrentFrame) &&
+                (this->skelAnime.animCurrentFrame < 43.0f)) {
```

Comment:
id flip 29 if possible

Reply:
Went with something a little different. As you know, I do prefer the ` a < x && x < b` ordering.